doc/source/tune/user-guide.rst

```diff
@@ -295,49 +295,157 @@ When running a hyperparameter search, Tune can automatically and periodically sa
  * use pre-emptible machines (by automatically restoring from last checkpoint)
  * Pausing trials when using Trial Schedulers such as HyperBand and PBT.
 
-To use Tune's checkpointing features, you must expose a ``checkpoint_dir`` argument in the function signature, and call ``tune.checkpoint_dir``:
+Let's cover how to configure your checkpoints storage location, checkpointing frequency, and how to resume from a previous run.
+
+A simple (cloud) checkpointing example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cloud storage-backed Tune checkpointing is the recommended best practice for both performance and reliability reasons. rsync-based checkpointing is also incompatible with Ray on Kubernetes. If you'd rather checkpoint locally or use rsync based checkpointing, see here.
+
+Prequisities to use cloud checkpointing in Ray Tune for the example below:
+
+Your ``my_trainable`` is either a:
+
+   1. **Model with an existing Ray integration**
+   
+      * XGBoost (:ref:`example <xgboost-ray-tuning>`)
+      * Pytorch (:ref:`example <tune-pytorch-lightning>`)
+      * Pytorch Lightning (:ref:`example <ray-lightning-tuning>`)
+      * Keras (:doc:`example </tune/examples/tune_mnist_keras>`)
+      * Tensorflow (:ref:`example <ray-train-tftrainer-example>`)
+      * LightGBM (:ref:`example <lightgbm-ray-tuning>`)
+  
+   2. **Custom training function**
+   
+      * All this means is that your function has to expose a ``checkpoint_dir`` argument in the function signature, and call ``tune.checkpoint_dir``. See :doc:`this example </tune/examples/custom_func_checkpointing>`, it's quite simple to do.
+
+Let's assume for this example you're running this script from your laptop, and connecting to your remote Ray cluster via ``ray.init()``, making your script on your laptop the "driver".
 
 .. code-block:: python
 
-    import os
-    import time
+    import ray
     from ray import tune
+    from your_module import my_trainable
+
+    ray.init(address="<cluster-IP>:<port>")  # set `address=None` to train on laptop
+
+    # configure how checkpoints are sync'd to the scheduler/sampler 
+    # we recommend cloud storage checkpointing as it survives the cluster when
+    # instances are terminated, and has better performance
+    sync_config = tune.syncConfig(
+        upload_dir="s3://my-checkpoints-bucket/path/",  # requires AWS credentials
+        sync_to_cloud=True,
+        sync_to_driver=False  # recommended for performance when using cloud checkpointing 
+    )
 
-    def train_func(config, checkpoint_dir=None):
-        start = 0
-        if checkpoint_dir:
-            with open(os.path.join(checkpoint_dir, "checkpoint")) as f:
-                state = json.loads(f.read())
-                start = state["step"] + 1
+    # here we wrap our Ray model (or custom training func) with a couple of things:
+    # 1) with_parameters() ensures when we checkpoint, weights are saved (not just hyperparameters)
```

Comment:
this isn't correct. `with_parameters` is used to pass additional constants to the tuning function, especially ones that are too large (eg. data). it has nothing to do with checkpointing

Reply:
got it. so why would one use that? and is simply `tune.durable(my_trainable)` good enough here?

Reply:
You'd use it in this case:
```
def trainable_func(conf, data):
    # code

tune.run(tune.with_parameters(trainable_func, data=data))
```

`data` will be put into the object store internally for efficient sharing

Reply:
Just FYI, `tune.durable()` will be deprecated this week, so might have to adjust docs then